packages/frontend/src/components/message/Message.tsx

```diff
@@ -1161,22 +1161,54 @@ function WebxdcMessageContent({
     )
 
     return cleanup
-  }, [
-    accountId,
-    message.id,
-    message.viewType,
-    fetchWebxdcInfo,
-    debouncedFetchWebxdcInfo,
-  ])
+  }, [accountId, message.id, debouncedFetchWebxdcInfo])
 
   if (message.viewType !== 'Webxdc') {
     return null
   }
 
-  const info = webxdcInfo || {
-    name: isLoadingWebxdcInfo ? 'Loading...' : 'INFO MISSING!',
+  type WebxdcInfo =
+    | T.WebxdcMessageInfo
+    | { name: string; document: undefined; summary: string }
+
+  const info: WebxdcInfo = webxdcInfo || {
+    name: fetchWebxdcInfo.loading ? 'Loading...' : 'INFO MISSING!',
     document: undefined,
-    summary: isLoadingWebxdcInfo ? '' : 'INFO MISSING!',
+    summary: fetchWebxdcInfo.loading ? '' : 'INFO MISSING!',
+  }
+
+  /**
+   * To avoid a dynamic height change which would cause wrong calculation in the
+   * message list height, we do not allow line breaks here. So we have to calculate
+   * an appropriate max length for the name and document depending on the window width.
+   * We can't use the messageWidth or containerWidth here because they are
+   * dynamic and depending on the length of the text itself.
+   */
+  const defaultMaxLength = 66
+  const getMaxLength = () => {
+    if (windowWidth > 1040) return defaultMaxLength
+    if (windowWidth > 760) return 46
+    if (windowWidth > 330) return 36
+    return 20
+  }
+  const maxLength = getMaxLength()
+
+  const truncateInfo = (info: WebxdcInfo) => {
+    if (!info.document || info.name.length >= maxLength - 4) {
+      // skip info.document if name is too long
+      return truncateText(info.name, maxLength)
+    }
+
+    let infoText = info.document + ' - ' + info.name
+
+    if (infoText.length > maxLength) {
+      infoText =
+        truncateText(info.document, maxLength - info.name.length) +
+        ' - ' +
+        info.name
+    }
+
+    return infoText
```

Comment:
Is this impossible to solve with CSS? Isn't this just "no linebreaks with ellipsis"?

Reply:
Well, the summary could be get a fix width with a number of media queries, but the document/name string should be treated with priority for the name, so the width of the document depends on the length of the name

Reply:
> but the document/name string should be treated with priority for the name

Would `flex-grow` / `flex-shrink` work for this? I haven't reviewed what the code does yet just so as you know.